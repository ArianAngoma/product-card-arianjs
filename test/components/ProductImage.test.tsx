import React from 'react';
import renderer from 'react-test-renderer';

/* Importaciones propias */
import {ProductImage, ProductCard} from '../../src';
// @ts-ignore
import {product2} from '../data/products';

describe('ProductImage', () => {
    test('Debe de mostrar en componente correctamente con la imagen', () => {
        const wrapper = renderer.create(
            <ProductImage img="https://sigdeletras.com/images/blog/202004_react_leaflet/react.png"
                          className="custom-class"/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('Debe de mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});