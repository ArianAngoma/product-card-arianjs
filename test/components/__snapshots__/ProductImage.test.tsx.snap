// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="ProductImage"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrar en componente correctamente con la imagen 1`] = `
<img
  alt="ProductImage"
  className="productImg custom-class"
  src="https://sigdeletras.com/images/blog/202004_react_leaflet/react.png"
/>
`;
